File: app/src/main/java/com/sampietro/NaiveAAC/activities/history/ToBeRecordedInHistory.java
package com.sampietro.NaiveAAC.activities.history;

import java.util.List;

/**
 * <h1>ToBeRecordedInHistory</h1>
 *
 * <p><b>ToBeRecordedInHistory</b> abstract class for registration in history class.</p>
 * Refer to <a href="https://www.html.it/pag/62281/interfacce-list-e-queue-in-java/">www.html.it</a>
 * by <a href="https://www.html.it/author/azoia/">Alessandro Zoia</a>
 *
 * @see History
 * @see ToBeRecordedInHistoryImpl
 * @version     1.3, 05/05/22
 */
public abstract class ToBeRecordedInHistory<T> {
    protected List<T> voicesToBeRecordedInHistory;


    public abstract void add(T voiceToBeRecordedInHistory);
    public abstract void delete(T voiceToBeRecordedInHistory);


    public abstract int getNumberOfVoicesToBeRecordedInHistory();
    public abstract List<T> getVoicesToBeRecordedInHistory();
}

File: app/src/main/java/com/sampietro/NaiveAAC/activities/history/ToBeRecordedInHistory.kt
package com.sampietro.NaiveAAC.activities.history

/**
 * <h1>ToBeRecordedInHistory</h1>
 *
 *
 * **ToBeRecordedInHistory** abstract class for registration in history class.
 * Refer to [www.html.it](https://www.html.it/pag/62281/interfacce-list-e-queue-in-java/)
 * by [Alessandro Zoia](https://www.html.it/author/azoia/)
 *
 * @see History
 *
 * @see ToBeRecordedInHistoryImpl
 *
 * @version     4.0, 09/09/2023
 */
abstract class ToBeRecordedInHistory<T> {
    @JvmField
    var voicesToBeRecordedInHistory: MutableList<T>? = null
    abstract fun add(voiceToBeRecordedInHistory: T)
    abstract fun delete(voiceToBeRecordedInHistory: T)
    abstract fun getNumberOfVoicesToBeRecordedInHistory(): Int
    abstract fun getListVoicesToBeRecordedInHistory(): MutableList<T>?
}